qiskit/circuit/qpy_serialization.py

```diff
@@ -1530,9 +1539,33 @@ def load(file_obj):
     which will read the contents of the qpy and return a list of
     :class:`~qiskit.circuit.QuantumCircuit` objects from the file.
 
+    With custom gate definitions containing overidden methods such as :
+
+    .. code-block:: python
+
+        from qiskit.circuit import Gate as QiskitGate
+
+        class Gate1(QiskitGate):
+            def validate_parameter(self, single_param):
+                if not isinstance(single_param,(int, float, str, bool, dict)):
+                    raise Exception(f"parameter of type {type(single_param)} not supported")
+                return single_param
+
+    use custom_dict parameter to pass this definition to the load method :
+
+    .. code-block:: python
+
+        from qiskit.circuit import qpy_serialization
+        #let the circuit with the custom gate be saved in a file circ
+
+        with gzip.open('circ.qpy.gz', 'rb') as fd:
+            circuit = qpy_serialization.load(fd, custom_gate_dict={"wait": Gate1})
+
     Args:
         file_obj (File): A file like object that contains the QPY binary
             data for a circuit
+        custom_gate_dict (dict) : A dict representing Custom gates defined in the
+            circuit with overridden class methods.
```

Comment:
We need to describe what the keys and values should be in the dictionary, and what expectations there are on them.

Reply:
- Okay will do so :+1: , but leaving them out was based on the fact that an example had been provided instructing about the changed usage.